### DISCUSSION TITLE
Ability to filter duplicated messages by signalpower

### DISCUSSION THREAD

**Discussion by laleft** [jvde-github/AIS-catcher#304] [Q&A]
Hi everyone,

I have this scenario: two stations, each running AIS-Catcher. One of them acts as a server, and the other sends messages via UDP.

![ais_1](https://github.com/user-attachments/assets/c3cee2c0-4dfe-4231-a070-e95922a1abdc)

The problem is that sometimes the ships are within the line of sight of both stations, so the AIS-Catcher working as a server receives the same message twice.

My question is if there is an option in the AIS-Catcher settings to transmit only one message and store it in the database. In my opinion, it could be the message with the maximum signal strength.

Any suggestions?

**Comment by jvde-github** [comment]
The server always update the contents in the database with the message that comes in last. There is a field in the ship table that shows which input channel provided this message. You can run separate webviewers for each of the input channels and the combination with the filter option.

Now we want to, in case of a duplicate, pick one message based on signal strength. I don't know how to do this to be honest. We can implement in the program but bit exotic.

Only way I can see is to send the messages first to a Python program that passes on a message if it not a duplicate or a duplicate with a higher signal level. If it is really important I would do it like that. 

Probably not helpful.

**Reply by laleft** [reply]
Hi, Jasper! Thank you for your reply. Yes, mine is a very specific case maybe. What I'm trying to figure out is how apps like Marinetraffic or Vesselfinder process messages from the  same vessel coming from multiple stations.

In mi case, I am developing an app to monitor traffic and I'm using two stations for that. But one of them is to extend the range only, so when a vessel is within range of the two stations, then messages are transmitted in duplicate (both in the JSON output and in the database).

I guess I'll make a middleware to filter the messages before they reach my app (I'm using websockets).

Thank you very much anyway. You are very kind.

**Reply by jvde-github** [reply]
One thought as well is to look at AIS dispatcher, it takes in a stream (or multiple I guess) and can remove duplicates and downsample the signal as well (e.g., one position update per minute or so). That might then not require any programming if you only want to remove duplicates but best to develop something yourself so you have full control.
